shards/rust/src/core.rs

```diff
@@ -555,15 +557,15 @@ unsafe extern "C" fn cancel_future_c_call<
 
 // Update the existing activate function to work with our new data structure
 unsafe extern "C" fn activate_future_c_call<
-  F: Future<Output = Result<R, &'static str>> + Send + 'static,
+  F: Future<Output = Result<R, FastError>> + Send + 'static,
   R: Into<ClonedVar>,
   C: Fn(),
 >(
   _context: *mut SHContext,
   arg2: *mut c_void,
 ) -> SHVar {
   let data = arg2 as *mut FutureCallData<F, C>;
-  let f = std::ptr::read(&(*data).future); // Move out the future
+  let f = (*data).future.take().unwrap_unchecked(); // Move out the future
```

Comment:
is this ok? cos before it would not `Drop`

Reply:
asyncActivate blocks so I made the data owned by the caller on the stackm easier to manage.
Then pass it here by ref through arg2 void* and take future